kernel/kernel-jammy-src/6.0/0005-support-for-dynamic-change-of-i2c-bus-clk-rate.patch

```diff
@@ -0,0 +1,172 @@
+From ce6faaedef68204e8b132bf44a0e4d2432dfb7ad Mon Sep 17 00:00:00 2001
+From: Arun-Prasad-V <arun.prasad.v@intel.com>
+Date: Mon, 4 Nov 2024 12:52:39 +0530
+Subject: [PATCH] Support for dynamic change of i2c bus-clk-rate
+
+Signed-off-by: Arun Prasad V <arun.prasad.v@intel.com>
+---
+ drivers/i2c/busses/i2c-tegra.c | 42 +++++++++++++++++++++++++++++++
+ drivers/i2c/i2c-core-base.c    | 45 ++++++++++++++++++++++++++++++++++
+ include/linux/i2c.h            |  5 ++++
+ 3 files changed, 92 insertions(+)
+
+diff --git a/drivers/i2c/busses/i2c-tegra.c b/drivers/i2c/busses/i2c-tegra.c
+index cd50436e5734..4269343266c2 100644
+--- a/drivers/i2c/busses/i2c-tegra.c
++++ b/drivers/i2c/busses/i2c-tegra.c
+@@ -1484,6 +1484,37 @@ static int tegra_i2c_xfer_msg(struct tegra_i2c_dev *i2c_dev,
+ 	return 0;
+ }
+ 
++static int tegra_i2c_change_clock_rate(struct tegra_i2c_dev *i2c_dev)
++{
++	u32 val;
++	int err;
++
++	val = I2C_CNFG_NEW_MASTER_FSM | I2C_CNFG_PACKET_MODE_EN |
++	      FIELD_PREP(I2C_CNFG_DEBOUNCE_CNT, 2);
++
++	if (i2c_dev->hw->has_multi_master_mode)
++		val |= I2C_CNFG_MULTI_MASTER_MODE;
++
++	i2c_writel(i2c_dev, val, I2C_CNFG);
++
++	if (i2c_dev->prod_list)
++		tegra_i2c_config_prod_settings(i2c_dev);
```

Comment:
If the function is not "void" then you need to check return value here and in 37

Reply:
these are "void" functions only.